File: integration-tests/src/test/java/dev/capylang/test/PrivateDataTest.java
package dev.capylang.test;

import org.junit.jupiter.api.Test;

import java.lang.reflect.Modifier;
import java.util.Arrays;

import static org.assertj.core.api.Assertions.assertThat;

class PrivateDataTest {
    @Test
    void underscorePrefixedDataCanBeUsedInsideModule() {
        assertThat(PrivateData.parseInt(7)).isEqualTo("ok:7");
        assertThat(PrivateData.parseConflict("x")).isEqualTo("internal:x");
    }

    @Test
    void underscorePrefixedDataIsGeneratedAsPrivateRecord() {
        var privateParseClass = Arrays.stream(PrivateData.class.getDeclaredClasses())
                .filter(clazz -> clazz.getSimpleName().equals("_Parse"))
                .findFirst()
                .orElseThrow();
        var parseClass = Arrays.stream(PrivateData.class.getDeclaredClasses())
                .filter(clazz -> clazz.getSimpleName().equals("Parse"))
                .findFirst()
                .orElseThrow();

        assertThat(Modifier.isPrivate(privateParseClass.getModifiers())).isTrue();
        assertThat(Modifier.isPrivate(parseClass.getModifiers())).isFalse();
    }
}

File: capy/src/e2e-cfun/js/dev/capylang/test/PrivateData.test.js
const test = require('node:test');
const { assert, generatedModule } = require('./_support');

test('PrivateData', () => {
    const privateData = generatedModule('dev/capylang/test/PrivateData.js');
    assert.equal(privateData.parseInt(7), 'ok:7');
    assert.equal(privateData.parseConflict('x'), 'internal:x');
    assert.equal(Object.prototype.hasOwnProperty.call(privateData, '_Parse'), false);
    assert.equal(Object.prototype.hasOwnProperty.call(privateData, 'Parse'), true);
});
